servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/CoreMessageSender.java

```diff
@@ -908,9 +768,8 @@ private void processSendWork()
                 int sentMsgSize = 0;
                 Exception sendException = null;
 
-                try
-                {
-                    delivery = sendLinkCurrent.delivery(deliveryTag.getDeliveryTag().getBytes());
+                try {
+                    delivery = sendLinkCurrent.delivery(deliveryTag.getDeliveryTag().getBytes(UTF_8));
```

Comment:
[SpotBugs P1] Reliance on default encoding: Found a call to a method which will perform a byte to String (or String to byte) conversion, and will assume that the default platform encoding is suitable.